pkg/bpmn_engine/engine_jobs_activated.go

```diff
@@ -18,16 +19,32 @@ type activatedJob struct {
 	processDefinitionKey     int64
 	elementId                string
 	createdAt                time.Time
+	scope                    variable_scope.VarScope
+	localScope               variable_scope.VarScope
 }
 
 // ActivatedJob represents an abstraction for the activated job
 // don't forget to call Fail or Complete when your task worker job is complete or not.
 type ActivatedJob interface {
-	ProcessInstance
+	// GetInstanceKey get instance key from processInfo
+	GetInstanceKey() int64
+	// GetCreatedAt  get job create time
+	GetCreatedAt() time.Time
+	// GetState get instance state
+	GetState() process_instance.State
 
 	// GetKey the key, a unique identifier for the job
 	GetKey() int64
 
+	// GetVariable the varaible from variable scope  include local scope
+	GetVariable(key string) interface{}
```

Comment:
Here I realized, we have a clash/issue in the current  ```main``` branch.
That's my fault. Accidently I let the interface ActivatedJob inherit from ProcessInstance,
but this makes no sense at all.
I need to refactor this. 
FYI: This GetVariable() function in the ActivatedJob interface will stay
No need to make a change from your side at this line.

Reply:
Hi @nitram509 "That's my fault. Accidently I let the interface ActivatedJob inherit from ProcessInstance,
but this makes no sense at all.
I need to refactor this."

Yeah, I remember I put `ProcessInstance` inherit from `ActivatedJob`. I put that because I saw the previous code uses `ProcessInstance`.

Reply:
> Here I realized, we have a clash/issue in the current `main` branch. That's my fault. Accidently I let the interface ActivatedJob inherit from ProcessInstance, but this makes no sense at all. I need to refactor this. FYI: This GetVariable() function in the ActivatedJob interface will stay No need to make a change from your side at this line.


Yes, ActivatedJob  no need to  inherit  from ProcessInstance.

On the other hand, I thinks it's better to let each job has its own variable context than using the processInstance's context.

![image](https://user-images.githubusercontent.com/3797268/190969085-16d3d608-7ec8-4c71-9f4b-35814df31eb4.png)

It's common to parallelize generata_update_msg and generata_alarm_msg , the task generata_update_msg will see the variable's middle status  in the execution of generata_alarm_msg task if some variable named name will be update servial times .
Of couse, it's a bad code style to relay on variable like this.

Another reason I like new variable context is to follow  camunda's doc which says:

"When the process instance enters a subprocess or an activity, a new scope is created. Activities in this scope can see all variables of this and of higher scopes (i.e. parent scopes). However, activities outside of this scope can not see the variables which are defined in this scope."

So I think it's better to let each job has its owne scope and propagated all varaible to its parente scope which maybe processInstance's scope when the job was completed.


That's  the reason I'm trying to introduce GetVariable function in ActivatedJob.
The new introduced GetVariable  function  will conflicted (or misleading) with the same function inherit   from ProcessInstance.


I will merge the code of main branch after you refactoring job.